lib/Sema/CSBindings.cpp

```diff
@@ -1656,9 +1656,11 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
   PO.PrintTypesForDebugging = true;
 
   out.indent(indent);
-  out << "(";
-  if (auto typeVar = getTypeVariable())
-    out << "$T" << typeVar->getImpl().getID() << " ";
+  if (CS.solverState) {
```

Comment:
Without this, the initial Type Variables, binding options, attributes etc print their $T_ twice like this: 
```
 $T0 [allows bindings to: lvalue, noescape] ($T0 [attributes: delayed] [with possible bindings: <empty>]) @ locator@0x13721b600 [OverloadedDeclRef@/Users/amritpankaur/test.swift:65:9]
``` 
because `ConstraintSystem::print` will print $T_ , followed by binding options via `TypeVariableType::Implementation::print` then print the rest via `CSBindings::dump`, printing the type variables twice. Then later, Potential Bindings print entirely via `CSBindings::dump` and require the $T_ to print from there.